x-pack/plugin/sql/qa/src/main/resources/null.csv-spec

```diff
@@ -0,0 +1,60 @@
+//
+// Null expressions
+//
+
+dateTimeOverNull
+SELECT YEAR(CAST(NULL AS DATE)) d;
+
+d:i
+null
+;
+
+addOfNull
+SELECT CAST(NULL AS INT) + CAST(NULL AS FLOAT) AS n;
+
+n:d
+null
+;
+
+
+divOfNull
+SELECT 5 / CAST(NULL AS FLOAT) + 10 AS n;
+
+n:d
+null
+;
+
+coalesceJustWithNull
+SELECT COALESCE(null, null, null) AS c;
+
+c
+null
+;
+
+coalesceWithFirstNullOfString
+SELECT COALESCE(null, 'first') AS c;
+
+c:s
+first
+;
+
+coalesceWithFirstNullOfNumber
+SELECT COALESCE(null, 123) AS c;
+
+c:i
+123
+;
+
+coalesceMixed
+SELECT COALESCE(null, 123, null, 321) AS c;
+
+c:i
+123
+;
+
+coalesceScalar
+SELECT COALESCE(null, ABS(123) + 1) AS c;
+
+c:i
+124
+;
```

Comment:
Can you add a test where the first value is non-null? (all tests with COALESCE assume there is at least one null value before the first non-null)

Reply:
Added (there's an optimization test that tries that but anyway).